crates/bin/node/src/main.rs

```diff
@@ -68,27 +72,94 @@ async fn main() -> Result<(), anyhow::Error> {
+                ).await?;
+                trace!(name: "grpc-listen", port = grpc_address.port());
+
+                let http_future = launch_http_server_task(
+                    app_state,
+                    env_variables.http_port,
+                );
+
+                tokio::select! {
+                    grpc_res = grpc_future => match grpc_res {
+                        Ok(()) => eprintln!("gRPC task should never return"),
+                        Err(err) => eprintln!("gRPC task failed: {}", err),
+                    },
+                    http_res = http_future => match http_res {
+                        Ok(()) => eprintln!("HTTP task should never return"),
+                        Err(err) => eprintln!("HTTP task failed: {}", err),
+                    },
+                    sig = tokio::signal::ctrl_c() => match sig {
+                        Ok(()) => info!("Servers terminated"),
+                        Err(err) => eprintln!("unable to listen for shutdown signal: {}", err)
+                    }
+                };
+            }
+        }
+        (true, false) => {
+            // Only gRPC enabled
+            info!("Starting gRPC server only");
+            let (grpc_address, grpc_future) = launch_tonic_server_task(
+                app_state,
+                env_variables.grpc_port,
+            ).await?;
+            trace!(name: "grpc-listen", port = grpc_address.port());
+
+            tokio::select! {
+                grpc_res = grpc_future => match grpc_res {
+                    Ok(()) => eprintln!("gRPC task should never return"),
+                    Err(err) => eprintln!("gRPC task failed: {}", err),
+                },
+                sig = tokio::signal::ctrl_c() => match sig {
+                    Ok(()) => info!("gRPC task terminated"),
+                    Err(err) => eprintln!("unable to listen for shutdown signal: {}", err)
+                }
+            };
+        }
+        (false, true) => {
+            // Only HTTP enabled
+            #[cfg(feature = "http")]
+            {
+                info!("Starting HTTP server only");
+                let http_future = launch_http_server_task(
+                    app_state,
+                    env_variables.http_port,
+                );
+
+                tokio::select! {
+                    http_res = http_future => match http_res {
+                        Ok(()) => eprintln!("HTTP task should never return"),
+                        Err(err) => eprintln!("HTTP task failed: {}", err),
+                    },
+                    sig = tokio::signal::ctrl_c() => match sig {
+                        Ok(()) => info!("HTTP task terminated"),
+                        Err(err) => eprintln!("unable to listen for shutdown signal: {}", err)
+                    }
+                };
+            }
+        }
+        (false, false) => {
+            panic!("At least one server feature (grpc or http) must be enabled");
```

Comment:
at the top of the file, add another one like this
```
#[cfg(not(any(feature = "starknet")))]
compile_error!("At least one liquidity feature should be provided during compilation");
```

but wiht `#[cfg(not(any(feature = "grpc", feature = "rest")))]` and a compile error explaining that you must use at least one of the two API solution

Reply:
This make it fail during compilation and not execution. Which is better